splitio/storage/pluggable.py

```diff
@@ -207,3 +207,170 @@ def put(self, split):
+        if prefix is not None:
+            self._prefix = prefix + "." + self._prefix
+            self._segment_till_prefix = prefix + "." + self._segment_till_prefix
+
+    def update(self, segment_name, to_add, to_remove, change_number=None):
+        """
+        Update a segment. Create it if it doesn't exist.
+
+        :param segment_name: Name of the segment to update.
+        :type segment_name: str
+        :param to_add: Set of members to add to the segment.
+        :type to_add: set
+        :param to_remove: List of members to remove from the segment.
+        :type to_remove: Set
+        """
+        try:
+            self._pluggable_adapter.add_items(self._prefix.format(segment_name=segment_name), to_add)
+            self._pluggable_adapter.remove_items(self._prefix.format(segment_name=segment_name), to_remove)
+            if change_number is not None:
+                self._pluggable_adapter.set(self._segment_till_prefix.format(segment_name=segment_name), change_number)
+        except Exception:
+            _LOGGER.error('Error updating segment storage')
+            _LOGGER.debug('Error: ', exc_info=True)
+
+    def set_change_number(self, segment_name, change_number):
+        """
+        Store a segment change number.
+
+        :param segment_name: segment name
+        :type segment_name: str
+        :param change_number: change number
+        :type segment_name: int
+        """
+        try:
+            self._pluggable_adapter.set(self._segment_till_prefix.format(segment_name=segment_name), change_number)
+        except Exception:
+            _LOGGER.error('Error updating segment change number')
+            _LOGGER.debug('Error: ', exc_info=True)
+
+    def get_change_number(self, segment_name):
+        """
+        Get a segment change number.
+
+        :param segment_name: segment name
+        :type segment_name: str
+
+        :return: change number
+        :rtype: int
+        """
+        try:
+            return self._pluggable_adapter.get(self._segment_till_prefix.format(segment_name=segment_name))
+        except Exception:
+            _LOGGER.error('Error fetching segment change number')
+            _LOGGER.debug('Error: ', exc_info=True)
+            return None
+
+    def get_segment_names(self):
+        """
+        Get list of segment names.
+
+        :return: list of segment names
+        :rtype: str[]
+        """
+        try:
+            keys = []
+            for key in self._pluggable_adapter.get_keys_by_prefix(self._prefix[:-self._SEGMENT_NAME_LEMNGTH]):
+                if key[-self._TILL_LENGTH:] != 'till':
+                    keys.append(key[len(self._prefix[:-self._SEGMENT_NAME_LEMNGTH]):])
+            return keys
+        except Exception:
+            _LOGGER.error('Error getting segments')
+            _LOGGER.debug('Error: ', exc_info=True)
+            return None
+
+    def get_keys(self, segment_name):
```

Comment:
I think we don't need `get_keys`, `get_segments_count` and `get_segments_keys_count` for this storage (idem for `RedisSegmentStorage`), because this data is not being sent by telemetry in consumer mode. I must check and update the spec.

Maybe, we can add a `TODO` comment on these methods to consider removing them before releasing the features, if they are actually not used anywhere in the code.

Reply:
InMemory has get_segments_count and get_segments_keys_count, so I think we need to keep them, I don't see get_keys though, so will commented it out.